python/ray/ml/examples/rl_serving_example.py

```diff
@@ -0,0 +1,97 @@
+import argparse
+import gym
+import numpy as np
+
+import ray
+from ray.ml.checkpoint import Checkpoint
+from ray.ml.config import RunConfig
+from ray.ml.predictors.integrations.rl.rl_serve_env import RLServeEnv
+from ray.ml.predictors.integrations.rl.rl_predictor import RLPredictor
+from ray.ml.tests.test_rl_predictor import _DummyPolicy
+from ray.ml.train.integrations.rl.rl_trainer import RLTrainer
+from ray.ml.result import Result
+from ray.rllib import RolloutWorker
+from ray.serve.model_wrappers import ModelWrapperDeployment
+from ray import serve
+from ray.tune.tuner import Tuner
+
+
+def train_rl_ppo_online(num_workers: int, use_gpu: bool = False) -> Result:
+    trainer = RLTrainer(
+        run_config=RunConfig(stop={"training_iteration": 15}),
+        scaling_config={
+            "num_workers": num_workers,
+            "use_gpu": use_gpu,
+        },
+        algorithm="PPO",
+        config={
+            "env": "CartPole-v0",
+            "framework": "tf",
+        },
+    )
+    tuner = Tuner(
+        trainer,
+        run_config=RunConfig(stop={"training_iteration": 15}),
+        _tuner_kwargs={"checkpoint_at_end": True, "checkpoint_freq": 1},
+    )
+    return tuner.fit()[0]
+
+
+def serve_rl_model(checkpoint: Checkpoint, name="RLModel") -> str:
+    serve.start(detached=True)
+    deployment = ModelWrapperDeployment.options(name=name)
+    deployment.deploy(RLPredictor, checkpoint)
+    return deployment.url
+
+
+def run_external_env(endpoint_uri: str, num_batches: int = 3):
+    sub_env = gym.make("CartPole-v0")
+
+    ev = RolloutWorker(
```

Comment:
actually why don't you create an RLServeEnv directly?
RolloutWorker is specifically used for training, and we probably don't want to encourage users to create these directly?

Reply:
just to understand things clearly, run_external_env() is the client that uses a serving RL policy right?

why don't you simply step() through the underlying Cartopole env directly in a while loop? and querying the endpoint_uri for actions yourself?

I feel like to keep this example simple and clear, we should involve as few RLlib pieces as possible. for example, no ExternalEnv, and definitely no RolloutWorker.

this RL cuj script I wrote a while back may be useful to you? https://github.com/anyscale/rl-end-to-end-tutorial/blob/master/jobs.py#L207-L218

Reply:
Hm, I actually think that the ExternalEnv is one of the key pieces for serving RL. But I agree, maybe it's out of scope for this particular example. I'll update it to match the experience in `rl_example.py`

Reply:
I understand what you mean, I think we need a different piece though, once connectors are in.
like a piece that runs obs and action processing, calls in-process policy, or over an HTTP connection, etc.